include/crm/lrmd.h

```diff
@@ -408,69 +474,88 @@ typedef struct lrmd_api_operations_s {
-     *       library (not via the executor connection). This means that it is based
-     *       on the local host environment, even if the executor connection is to a
-     *       remote node, so (for most resource agent classes) this will fail if
-     *       the agent is not installed locally. This also means that, if an
-     *       external agent must be executed, it will be executed by the
-     *       caller's user, not the executor's.
-     * \todo Add a metadata call to the executor API and let the server handle this.
+     * \note Caller is responsible for freeing output. This call is always
+     *       synchronous (blocking), and always done directly by the library
+     *       (not via the executor connection). This means that it is based on
+     *       the local host environment, even if the executor connection is to a
+     *       remote node, so this may fail if the agent is not installed
+     *       locally. This also means that, if an external agent must be
+     *       executed, it will be executed by the caller's user, not the
+     *       executor's.
      */
-    int (*get_metadata) (lrmd_t * lrmd,
-                         const char *standard,
-                         const char *provider,
-                         const char *agent, char **output, enum lrmd_call_options options);
+    int (*get_metadata) (lrmd_t *lrmd, const char *standard,
+                         const char *provider, const char *agent,
+                         char **output, enum lrmd_call_options options);
 
     /*!
-     * \brief Retrieve a list of installed resource agents.
+     * \brief Retrieve a list of installed resource agents
+     *
+     * \param[in]  lrmd      Executor connection (unused)
+     * \param[out] agents    Where to store agent list (must not be NULL)
+     * \param[in]  standard  Resource agent standard to list
+     * \param[in]  provider  Resource agent provider to list (or NULL)
      *
      * \return Number of items in list on success, negative legacy Pacemaker
      *         return code otherwise
      *
      * \note if standard is not provided, all known agents will be returned
      * \note list must be freed using lrmd_list_freeall()
      */
-    int (*list_agents) (lrmd_t * lrmd, lrmd_list_t ** agents,
+    int (*list_agents) (lrmd_t *lrmd, lrmd_list_t **agents,
                         const char *standard, const char *provider);
 
     /*!
      * \brief Retrieve a list of resource agent providers
      *
+     * \param[in]  lrmd       Executor connection (unused)
+     * \param[in]  agent      If not NULL, list providers for this agent only
+     * \param[out] providers  Where to store provider list
+     *
      * \return Number of items in list on success, negative legacy Pacemaker
      *         return code otherwise
      *
-     * \note When the agent is provided, only the agent's provider will be returned
+     * \note When agent is provided, only provider will be returned.
```

Comment:
Something like "only the given agent's providers" reads more clearly to me -- "provider" by itself sounds like it refers to a particular variable we've defined previously. And there might be more than one provider, right? E.g., `"heartbeat"` and `"pacemaker"` when `agent` is `"Dummy"`.

With that said, both of the notes feel kind of redundant. The param documentation is clear enough to me.

Reply:
agreed